### ISSUE TITLE
Helm 2.15.1 deletes service account before pre-delete hook runs

### ISSUE THREAD

**Issue by yehiyam** [helm/helm#6799]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:18:23Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
microk8s

First seen with 2.15.1 and kubernetes 1.16. Worked OK with helm v2.14.3 abd kubernetes 1.12 to 1.14

In our chart we have a pre-delete hook that uses a service-account
when running ```helm delete --purge my-release```:
Previously, the service account was deleted only after the pre-delete hook pod was created.
Now, when creating the pre-delete pod, there is an error:
```
MountVolume.SetUp failed for volume "cleanup-serviceaccount-token-s5ct9" : secret "cleanup-serviceaccount-token-s5ct9" not found
```

**Comment by thomastaylor312** [comment]
So many of these reported hook issues may all be resolved by a recently merged fix #6897. Can anyone give the `dev-2` branch a try and see if it fixes it for them?

**Comment by bacongobbler** [comment]
While I don't have a chart which I can use to verify the behaviour locally, I'm fairly certain this has been fixed in #6897. That patch fixes an issue with the kube client unable to wait for Jobs to run to completion as seen in #6894. Kubernetes 1.16 introduced a breaking change to the legacy schema so we had to update the code to reflect that change.

I'm going to close this as tentatively resolved via #6897, which will be in Helm 2.16.1. Please feel free to report your findings once 2.16.1 has been released with the likely fix mentioned above and we'll re-open if this is still an issue in 2.16.1+. I'm interested to hear your findings!

**Comment by thomastaylor312** [comment]
I am pretty certain that this should be solved by #6907. @yehiyam you are correct in that it wasn't waiting for certain jobs (see the explanation in the linked PR). Going to leave this closed for now as the PR is now open. If that PR does not resolve the issue, please let us know

**Comment by yehiyam** [comment]
@thomastaylor312 Thanks. I'll try it when 2.16.1 is released
